src/main/java/org/jabref/logic/integrity/IntegrityIssue.java

```diff
@@ -0,0 +1,72 @@
+    NO_INTEGER_AS_VALUE_FOR_EDITION_ALLOWED(Localization.lang("no integer as values for edition allowed")),
+    SHOULD_CONTAIN_AN_INTEGER_OR_A_LITERAL(Localization.lang("should contain an integer or a literal")),
+    SHOULD_HAVE_THE_FIRST_LETTER_CAPITALIZED(Localization.lang("should have the first letter capitalized")),
+    REFERENCED_CITATION_KEY_DOES_NOT_EXIST(Localization.lang("Referenced citation key does not exist")),
+    NON_ASCII_ENCODED_CHARACTER_FOUND(Localization.lang("Non-ASCII encoded character found")),
+    SHOULD_CONTAIN_A_VALID_PAGE_NUMBER_RANGE(Localization.lang("should contain a valid page number range")),
+    CAPITAL_LETTER_ARE_NOT_MASKED_USING_CURLY_BRACKETS(Localization.lang("capital letters are not masked using curly brackets {}"), Localization.lang("Mask letters")),
+    BOOKTITLE_ENDS_WITH_CONFERENCE_ON(Localization.lang("booktitle ends with 'conference on'")),
+    INCORRECT_FORMAT(Localization.lang("incorrect format")),
+    INCORRECT_CONTROL_DIGIT(Localization.lang("incorrect control digit")),
+    ABBREVIATION_DETECTED(Localization.lang("abbreviation detected")),
+    ODD_NUMBER_OF_UNESCAPED(Localization.lang("odd number of unescaped '#'")),
+    ENTRY_TYPE_IS_ONLY_DEFINED_FOR_BIBLATEX_BUT_NOT_FOR_BIBTEX(Localization.lang("Entry type %0 is only defined for Biblatex but not for BibTeX")),
+    UNEXPECTED_OPENING_CURLY_BRACKET(Localization.lang("unexpected opening curly bracket")),
+    UNEXPECTED_CLOSING_CURLY_BRACKET(Localization.lang("unexpected closing curly bracket")),
+    DUPLICATE_CITATION_KEY(Localization.lang("Duplicate citation key")),
+    SAME_DOI_USED_IN_MULTIPLE_ENTRIES(Localization.lang("Same DOI used in multiple entries")),
+    DOI_IS_INVALID(Localization.lang("DOI is invalid")),
+    LINK_SHOULD_REFER_TO_A_CORRECT_FILE_PATH(Localization.lang("link should refer to a correct file path")),
+    SHOULD_HAVE_THE_FIRST_CHARACTER_CAPITALIZED(Localization.lang("should have the first letter capitalized")),
+    HTML_ENCODED_CHARACTER_FOUND(Localization.lang("HTML encoded character found")),
+    LAST_FOUR_NONPUNCTUATION_CHARACTERS_SHOULD_BE_NUMERALS(Localization.lang("last four nonpunctuation characters should be numerals")),
+    SHOULD_CONTAIN_A_FOUR_DIGIT_NUMBER(Localization.lang("should contain a four digit number")),
+    EMPTY_CITATION_KEY(Localization.lang("empty citation key")),
+    INVALID_CITATION_KEY(Localization.lang("Invalid citation key")),
+    SHOULD_BE_NORMALIZED(Localization.lang("should be normalized")),
+    SHOULD_BE_AN_INTEGER_OR_NORMALIZED(Localization.lang("should be an integer or normalized"));
+
+    private final String text;
+    private final String fix;
+
+    IntegrityIssue(String text, String fix) {
+        this.text = text;
+        this.fix = fix;
+    }
+
+    IntegrityIssue(String text) {
+        this.text = text;
+        this.fix = null;
+    }
+
+    public String getText() {
+        return text;
+    }
+
+    public String getFix() {
+        return fix;
```

Comment:
New public methods should not return `null`. They should make use of `java.util.Optional`. In case `null` really needs to be used, the [JSpecify](https://jspecify.dev/) annotations must be used.